api/models/itemDao.js

```diff
@@ -115,22 +125,45 @@ const getTeacups = async (sorting = 'id', price = null, category) => {
                   AND category_id = ${CATEGORY_ID[category]}
                     ${priceClause}
                 GROUP BY id ${orderClause}
-        `
-    );
+        `);
 
-
-    const categoryData = await appDataSource.query(`
+  const categoryData = await appDataSource.query(`
         SELECT category_description.category_title,
                category_description.category_description
         FROM category_description
                  INNER JOIN item_types ON category_description.item_type_id = item_types.id
                  INNER JOIN categories ON categories.id = category_description.categories_id
         WHERE item_type_id = ${TYPE_ID["teabags"]}
-          AND categories_id = ${CATEGORY_ID[category]}`)
+          AND categories_id = ${CATEGORY_ID[category]}`);
 
-    return [itemData, categoryData];
+  return [itemData, categoryData];
 };
 
+const getDetailItem = async (itemId) => {
+  return appDataSource.query(
+    `SELECT
+    i.id,
+    i.name,
+    categories.category_name AS category_name,
+    i.description AS description,
+    item_types.type_name AS type_name,
+    i.image_url,
+    ROUND(i.price) AS price,
+    item_sizes.teabag_size AS teabag_size,
+    JSON_ARRAYAGG(
+    tasting_notes.note_name) AS tasting_notes
+    FROM items AS i
+    INNER JOIN item_types ON  i.item_type_id=item_types.id
+    INNER JOIN categories ON i.category_id=categories.id
+    LEFT JOIN item_sizes ON i.size_id=item_sizes.id
+    LEFT JOIN items_tasting_notes ON i.id=items_tasting_notes.item_id
+    LEFT JOIN tasting_notes ON items_tasting_notes.tasting_note_id=tasting_notes.id
+    WHERE i.id=?`,
+    [itemId]
+  );
+};
```

Comment:
보통 상품의 상세 데이터를 위한 dao 함수라면 return되는 객체는 단 하나일 것이기 때문에, 배열을 제거 해서 보내주는것이 나을 것 같습니다!
```suggestion
const getDetailItem = async (itemId) => {
  const [result] = await appDataSource.query(
    `SELECT
    i.id,
    i.name,
    categories.category_name AS category_name,
    i.description AS description,
    item_types.type_name AS type_name,
    i.image_url,
    ROUND(i.price) AS price,
    item_sizes.teabag_size AS teabag_size,
    JSON_ARRAYAGG(
    tasting_notes.note_name) AS tasting_notes
    FROM items AS i
    INNER JOIN item_types ON  i.item_type_id=item_types.id
    INNER JOIN categories ON i.category_id=categories.id
    LEFT JOIN item_sizes ON i.size_id=item_sizes.id
    LEFT JOIN items_tasting_notes ON i.id=items_tasting_notes.item_id
    LEFT JOIN tasting_notes ON items_tasting_notes.tasting_note_id=tasting_notes.id
    WHERE i.id=?`,
    [itemId]
  );
  
  return result
};
```